src/sysupdate/sysupdate-transfer.c

```diff
@@ -466,18 +460,42 @@ static int config_parse_partition_flags(
         return 0;
 }
 
-int transfer_read_definition(Transfer *t, const char *path) {
-        int r;
+static bool decide_if_enabled(Hashmap *known, char **regular, char **requisite) {
+        /* Requisite feature disabled -> transfer disabled */
+        STRV_FOREACH(id, requisite) {
+                Feature *f = hashmap_get(known, *id);
+                if (!f || !f->enabled) /* missing features are implicitly disabled */
+                        return false;
+        }
+
+        /* No features defined -> transfer implicitly enabled */
+        if (strv_isempty(regular))
+                return true;
+
+        /* At least one feature enabled -> transfer enabled */
+        STRV_FOREACH(id, regular) {
+                Feature *f = hashmap_get(known, *id);
+                if (f && f->enabled)
+                        return true;
+        }
+
+        /* All listed features disabled -> transfer disabled */
+        return false;
+}
+
+int transfer_read_definition(Transfer *t, const char *path, const char **dirs, Hashmap *known_features) {
+        _cleanup_strv_free_ char **features = NULL, **req_features = NULL;
 
         assert(t);
-        assert(path);
 
         ConfigTableItem table[] = {
                 { "Transfer",    "MinVersion",              config_parse_min_version,          0, &t->min_version             },
                 { "Transfer",    "ProtectVersion",          config_parse_protect_version,      0, &t->protected_versions      },
                 { "Transfer",    "Verify",                  config_parse_bool,                 0, &t->verify                  },
                 { "Transfer",    "ChangeLog",               config_parse_url_specifiers,       0, &t->changelog               },
                 { "Transfer",    "AppStream",               config_parse_url_specifiers,       0, &t->appstream               },
+                { "Transfer",    "Features",                config_parse_strv,                 0, &features                   },
+                { "Transfer",    "RequisiteFeatures",       config_parse_strv,                 0, &req_features               },
```

Comment:
hmm, i think we should store these two in Transfer as well. Yes, we only need them early on, but still, not a fan of such "optimizations". It# so little memory, and it makes the description more complete and debuggable. Hence, just add two fiels to Transfer for this.